### ISSUE TITLE
Enable "hot" codeready workspaces

### ISSUE THREAD

**Issue by andykrohg** [jharmison-redhat/openshift-devsecops#37]
The CRW experience can be significantly improved by
1. Creating workspaces ahead of time
```yaml
---
- name: "Get Che {{ user }} token"
  uri:
    url: http://keycloak-codeready.{{ route_subdomain }}/auth/realms/codeready/protocol/openid-connect/token
    method: POST
    body:
      username: "{{ user }}"
      password: "{{ workshop_che_user_password }}"
      grant_type: "password"
      client_id: "admin-cli"
    body_format: form-urlencoded
    status_code: 200
  register: user_token

- name: Create workspace for {{ user }} from devfile
  uri:
    url: "http://codeready-codeready.{{ route_subdomain }}/api/workspace/devfile?start-after-create=true&namespace={{ user }}"
    method: POST
    headers:
      Content-Type: application/json
      Authorization: "Bearer {{ user_token.json.access_token }}"
    body: "{{ lookup('template', './templates/devfile.json.j2') }}"
    body_format: json
    status_code: 201,409
  register: workspace_def
```
2. Removing the workspace idle timeout (default is 30 minutes)
```yaml
---
apiVersion: org.eclipse.che/v1
kind: CheCluster
spec:
  server:
    customCheProperties:
      CHE_LIMITS_WORKSPACE_IDLE_TIMEOUT: "0"
```
3. Extending the lifespan of keycloak tokens
```yaml
- name: Get codeready SSO admin token
  uri:
    url: http://keycloak-codeready.{{ route_subdomain }}/auth/realms/master/protocol/openid-connect/token
    method: POST
    body:
      username: "{{ codeready_sso_admin_username }}"
      password: "{{ codeready_sso_admin_password }}"
      grant_type: "password"
      client_id: "admin-cli"
    body_format: form-urlencoded
    status_code: 200,201,204
  register: codeready_sso_admin_token

- name: Increase codeready access token lifespans
  uri:
    url: http://keycloak-codeready.{{ route_subdomain }}/auth/admin/realms/codeready
    method: PUT
    headers:
      Content-Type: application/json
      Authorization: "Bearer {{ codeready_sso_admin_token.json.access_token }}"
    body:
      accessTokenLifespan: 28800
      accessTokenLifespanForImplicitFlow: 28800
      actionTokenGeneratedByUserLifespan: 28800
      ssoSessionIdleTimeout: 28800
      ssoSessionMaxLifespan: 28800
    body_format: json
    status_code: 204
```

**Comment by jharmison-redhat** [comment]
Have the basis of this done on the [hotload_crw branch](https://github.com/jharmison-redhat/openshift-devsecops/tree/hotload_crw), just need a devfile template now and can implement the last part.

**Comment by andykrohg** [comment]
Also pre-create OpenShift principals in keycloak

**Comment by jharmison-redhat** [comment]
Closed in #59
